src/main/java/com/commercetools/sync/services/impl/CustomObjectServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+package com.commercetools.sync.services.impl;
+
+import com.commercetools.sync.commons.BaseSyncOptions;
+import com.commercetools.sync.commons.models.NonResolvedReferencesCustomObject;
+import com.commercetools.sync.services.CustomObjectService;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.CustomObjectDraft;
+import io.sphere.sdk.customobjects.commands.CustomObjectDeleteCommand;
+import io.sphere.sdk.customobjects.commands.CustomObjectUpsertCommand;
+import io.sphere.sdk.customobjects.queries.CustomObjectQuery;
+import io.sphere.sdk.queries.PagedResult;
+import io.sphere.sdk.queries.QueryPredicate;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.util.Collections.singleton;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+public class CustomObjectServiceImpl
+        implements CustomObjectService {
+
+    private final BaseSyncOptions syncOptions;
+
+    private static final String CREATE_FAILED = "Failed to create draft with key: '%s'. Reason: %s";
+
+    public CustomObjectServiceImpl(final BaseSyncOptions baseSyncOptions) {
+        this.syncOptions = baseSyncOptions;
+    }
+
+    @Nonnull
+    @Override
+    public CompletionStage<Optional<CustomObject<NonResolvedReferencesCustomObject>>>
+        fetchCustomObject(@Nullable final String key) {
+
+        if (isBlank(key)) {
+            return CompletableFuture.completedFuture(Optional.empty());
+        }
+
+        final CustomObjectQuery<NonResolvedReferencesCustomObject> customObjectQuery = CustomObjectQuery
+                .of(NonResolvedReferencesCustomObject.class)
+                .withPredicates(buildCustomObjectKeysQueryPredicate(singleton(key)));
```

Comment:
Consider using `CustomObjectByKeyGet`
```java
        return syncOptions
            .getCtpClient()
            .execute(CustomObjectByKeyGet
                .of(CUSTOM_OBJECT_CONTAINER_KEY, key, NonResolvedReferencesCustomObject.class))
            .thenApply(Optional::ofNullable);
```

Reply:
Then again we have to answer the same question, what do we consider to be the `container` of the COs we create through this service?

Reply:
Please ignore the above comment. I have the answer in below 👍

Reply:
a081501113b7ca7fa8c716a7491c56a081edb0f0